tests/next_tests/unit_tests/embedded_tests/test_nd_array_field.py

```diff
@@ -125,3 +131,54 @@ def fma(a: common.Field, b: common.Field, c: common.Field, /) -> common.Field:
 
     result = fma(field_inp_a, field_inp_b, field_inp_c)
     assert np.allclose(result.ndarray, expected)
+
+
+def test_get_slices_with_named_indices_1d_to_2d_missing_dim_right():
+    field_domain = common.Domain(dims=(IDim,), ranges=(UnitRange(0, 10),))
+    new_domain = common.Domain(dims=(IDim, JDim), ranges=(UnitRange(5, 10), UnitRange(5, 10)))
+    field = common.field(np.ones((10,)), domain=field_domain)
+    slices = _get_slices_with_named_indices(field, new_domain)
+    assert slices == (slice(5, 10, None), None)
+
+
+def test_get_slices_with_named_indices_1d_to_2d_missing_dim_left():
+    field_domain = common.Domain(dims=(JDim,), ranges=(UnitRange(0, 10),))
+    new_domain = common.Domain(dims=(IDim, JDim), ranges=(UnitRange(0, 5), UnitRange(0, 10)))
+    field = common.field(np.ones((10,)), domain=field_domain)
+    slices = _get_slices_with_named_indices(field, new_domain)
+    assert slices == (None, slice(0, 10, None))
+
+
+def test_get_slices_with_named_indices_1d_to_3d():
+    field_domain = common.Domain(dims=(IDim,), ranges=(UnitRange(0, 10),))
+    new_domain = common.Domain(dims=(IDim, JDim, KDim), ranges=(UnitRange(0, 5), UnitRange(0, 10), UnitRange(0, 10)))
+    field = common.field(np.ones((10,)), domain=field_domain)
+    slices = _get_slices_with_named_indices(field, new_domain)
+    assert slices == (slice(0, 5, None), None, None)
+
+
+def test_get_slices_with_named_indices_3d_to_1d():
+    field_domain = common.Domain(dims=(IDim, JDim, KDim), ranges=(UnitRange(0, 10), UnitRange(0, 10), UnitRange(0, 10)))
+    new_domain = common.Domain(dims=(IDim,), ranges=(UnitRange(0, 10),))
+    field = common.field(np.ones((10, 10, 10)), domain=field_domain)
+    slices = _get_slices_with_named_indices(field, new_domain)
+    assert slices == (slice(0, 10, None),)
```

Comment:
This is wrong (or maybe correct by chance). What do you expect the result to be if you apply this slice to a 3D field. What happens if `new_domain` has JDim instead of IDim?

For safety I would always produce N-tuple for slicing when `shape(field)==N` to make sure we don't accidentally broadcast in the wrong dimension.

Reply:
Ok, actually it's more complicated and maybe it makes sense to first write down what we expect to happen.

Reply:
Not sure I understand why